local/compose/create.go

```diff
@@ -180,6 +180,9 @@ func (s *composeService) getCreateOptions(ctx context.Context, p *types.Project,
 	labels[workingDirLabel] = p.WorkingDir
 	labels[configFilesLabel] = strings.Join(p.ComposeFiles, ",")
 	labels[containerNumberLabel] = strconv.Itoa(number)
+	if len(service.EnvFile) > 0 {
+		labels[environmentFilesLabel] = strings.Join(service.EnvFile, ",")
```

Comment:
You're storing the service's env_file attribute in label doing so, while the expected value is the one set by user as `--env-file` on command line (if present)
see 
-> https://github.com/docker/compose/blob/master/compose/cli/command.py#L39
-> https://github.com/docker/compose/blob/master/compose/cli/command.py#L162
-> https://github.com/docker/compose/blob/master/compose/cli/command.py#L177

Reply:
Good catch!